### PR TITLE
[communication] Add MicrosoftTeamsUser identifier kind and make id optional

### PR DESCRIPTION
**Pull request by DominikMe**
also fixes two missing types

### CODE REVIEW

**Review by RezaJooyandeh — APPROVED** [review]

**Comment by RezaJooyandeh** [comment]
👋🏻 @tjprescott, this is still a preview swagger. Do you know why the pipeline is asking us to fill the breaking change template?

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
Please elaborate on the breaking change. Is this a change, or fixing a "bug" in the Swagger?

**Review by amrElroumy — APPROVED** [review]

File: specification/communication/data-plane/Microsoft.CommunicationServicesCommon/preview/2020-11-19-preview1/common.json [hunk 1]
```diff
@@ -63,18 +65,18 @@
...
+        "callingApplication",
+        "microsoftTeamsUser"
       ],
       "x-ms-enum": {
         "name": "CommunicationIdentifierKind",
         "modelAsString": true
       }
     },
-    "CommunicationIdentifier": {
+    "CommunicationIdentifierModel": {
```
Comment (amrElroumy) [inline specification/communication/data-plane/Microsoft.CommunicationServicesCommon/preview/2020-11-19-preview1/common.json:76]: whould we add `type` for this as well?

Comment (DominikMe) [reply specification/communication/data-plane/Microsoft.CommunicationServicesCommon/preview/2020-11-19-preview1/common.json:76]: yes, good catch

**Comment by DominikMe** [comment]
> Please elaborate on the breaking change. Is this a change, or fixing a "bug" in the Swagger?

@tjprescott The breaking change that was discovered by CI is the addition of `"type": "object"` to the model definitions. I had forgot to add them, and autorest had automatically inferred type: object with a Warning.

This common swagger hasn't been used in any service yet and we are adjusting it now while integrating it with the new versions for our various communication services.

**Comment by tjprescott** [comment]
Thanks @DominikMe. @lmazuel can you merge this PR?

**Review by lmazuel — APPROVED** [review]

**Pull request merged**